REFERENCE.md

```diff
@@ -0,0 +1,493 @@
+       '192.168.34.62',
+     ],
+     'ports'       => {
+       'rpc'   => 8567,
+       'https' => 8500,
+       'http'  => -1,
+     },
+   }
+
+   -----------------
+   -- SORTED JSON --
+   -----------------
+
+   nomad_sorted_json(unsorted_hash)
+
+   {"bind_addr":"192.168.34.56","client_addr":"127.0.0.1",
+   "ports":{"http":-1,"https":8500,"rpc":8567},
+   "start_join":["192.168.34.60","192.168.34.61","192.168.34.62"]}
+
+   ------------------------
+   -- PRETTY SORTED JSON --
+   ------------------------
+   Params: data <hash>, pretty <true|false>, indent <int>.
+
+   nomad_sorted_json(unsorted_hash, true, 4)
+
+   {
+       "bind_addr": "192.168.34.56",
+       "client_addr": "127.0.0.1",
+       "ports": {
+           "http": -1,
+           "https": 8500,
+           "rpc": 8567
+       },
+       "start_join": [
+           "192.168.34.60",
+           "192.168.34.61",
+           "192.168.34.62"
+       ]
+   }
+
+#### `nomad::nomad_sorted_json(Any *$args)`
+
+---- original file header ----
+
+Optionally you can pass 2 additional parameters, pretty generate and indent length.
+
+*Examples:*
+
+   -------------------
+   -- UNSORTED HASH --
+   -------------------
+   unsorted_hash = {
+     'client_addr' => '127.0.0.1',
+     'bind_addr'   => '192.168.34.56',
+     'start_join'  => [
+       '192.168.34.60',
+       '192.168.34.61',
+       '192.168.34.62',
+     ],
+     'ports'       => {
+       'rpc'   => 8567,
+       'https' => 8500,
+       'http'  => -1,
+     },
+   }
+
+   -----------------
+   -- SORTED JSON --
+   -----------------
+
+   nomad_sorted_json(unsorted_hash)
+
+   {"bind_addr":"192.168.34.56","client_addr":"127.0.0.1",
+   "ports":{"http":-1,"https":8500,"rpc":8567},
+   "start_join":["192.168.34.60","192.168.34.61","192.168.34.62"]}
+
+   ------------------------
+   -- PRETTY SORTED JSON --
+   ------------------------
+   Params: data <hash>, pretty <true|false>, indent <int>.
+
+   nomad_sorted_json(unsorted_hash, true, 4)
+
+   {
+       "bind_addr": "192.168.34.56",
+       "client_addr": "127.0.0.1",
+       "ports": {
+           "http": -1,
+           "https": 8500,
+           "rpc": 8567
+       },
+       "start_join": [
+           "192.168.34.60",
+           "192.168.34.61",
+           "192.168.34.62"
+       ]
+   }
+
+Returns: `Data type` Describe what the function returns here
+
+##### `*args`
+
+Data type: `Any`
+
+The original array of arguments. Port this to individually managed params
+to get the full benefit of the modern function API.
+
+### `nomad::sorted_json`
+
+Type: Ruby 4.x API
+
+This function takes unsorted hash and outputs JSON object making sure the keys are sorted.
+
+#### `nomad::sorted_json(Optional[Any] $unsorted_hash = {}, Optional[Any] $pretty = false, Optional[Any] $indent_len = 4)`
+
+The nomad::sorted_json function.
+
+Returns: `Hash`
+
+##### `unsorted_hash`
+
+Data type: `Optional[Any]`
+
+
+
+##### `pretty`
+
+Data type: `Optional[Any]`
```

Comment:
Would be nice if there was a clearer data type for this and the length.They could also use descriptions.

Reply:
ok ill add them